spec/requests/api/v2/providers/courses/publishable_spec.rb

```diff
@@ -69,8 +49,9 @@
     describe "failed validation" do
       let(:json_data) { JSON.parse(subject.body)["errors"] }
 
-      context "no enrichments" do
-        let(:course) { create(:course, provider: provider) }
+      context "no enrichments and location" do
+        let(:course) { TestSetup.unpublished_course_with_no_location_or_enrichment }
```

Comment:
How 'granular' should we be with regards to course test data?

Reply:
I think we should increase granularity as needed rather than upfront so as to try and have a more uniform set of test courses, but I do think some tests will require high degrees of granularity although I can't think of any off the top of my head.

Reply:
I think we should try to list out what courses we could reasonably expect to see in the wild. For stuff like failure case testing, maybe a single course that doesn't have any values set might be enough ... trying to cater to each test's requirements with a TestSetup course would be overkill.

Reply:
I'm also a bit concerned about the proliferation of methods like this for every variation. I think we should set a strong opinion out as to what in what cases `TestSetup` should be extended vs. just building custom things. `unpublished_course_with_no_location_or_enrichment` smells to me a bit of heading down the proliferation path already and maybe we shouldn't change this one?

Reply:
Ya, I agree that this one sounds a bit too specialised.